test-curves/src/bn384_small_two_adicity/g1.rs

```diff
@@ -0,0 +1,47 @@
+use ark_ec::{
+    models::{ModelParameters, SWModelParameters},
+    short_weierstrass_jacobian::*,
+};
+use ark_ff::{field_new, Zero};
+
+use crate::bn384_small_two_adicity::{Fq, Fr};
+
+pub type G1Affine = GroupAffine<Parameters>;
+pub type G1Projective = GroupProjective<Parameters>;
+
+#[derive(Clone, Default, PartialEq, Eq)]
+pub struct Parameters;
+
+impl ModelParameters for Parameters {
+    type BaseField = Fq;
+    type ScalarField = Fr;
+}
+
+impl SWModelParameters for Parameters {
+    /// COEFF_A = 0
+    const COEFF_A: Fq = field_new!(Fq, "0");
+
+    /// COEFF_B = 3
+    const COEFF_B: Fq = field_new!(Fq, "17");
+
+    /// COFACTOR = 1
+    const COFACTOR: &'static [u64] = &[0x1];
+
+    /// COFACTOR_INV = COFACTOR^{-1} mod r = 1
+    const COFACTOR_INV: Fr = field_new!(Fr, "1");
+
+    /// AFFINE_GENERATOR_COEFFS = (G1_GENERATOR_X, G1_GENERATOR_Y)
+    const AFFINE_GENERATOR_COEFFS: (Self::BaseField, Self::BaseField) =
+        (G1_GENERATOR_X, G1_GENERATOR_Y);
+
+    #[inline(always)]
+    fn mul_by_a(_: &Self::BaseField) -> Self::BaseField {
+        Self::BaseField::zero()
+    }
+}
+
+/// G1_GENERATOR_X = 1
+pub const G1_GENERATOR_X: Fq = field_new!(Fq, "1");
+
+/// G1_GENERATOR_Y = 2
+pub const G1_GENERATOR_Y: Fq = field_new!(Fq, "1564103007291248495620802981106927637056875497719363782670466283602827039808131416274891289504983049472454474542799");
```

Comment:
Why is this generator's y-coordinate so large?

Reply:
So this is because b=17 is the first value that satisfies the criteria (note: if b=3 satisfies we would have a simple generator, but nope). And the y coordinate here is the sqrt root of 18.

Reply:
And I think clearly I should fix the comment which says 2 and also add a note that this is sqrt 18

Reply:
updated

Reply:
Maybe we can set x = -1? That way y = 4

Reply:
Yes. You are right. It is valid.

Reply:
Wait...let me check the order of the point more carefully :)

Reply:
Yes.